dev-tools/docker/linux_image/Dockerfile

```
@@ -82,3 +90,51 @@ RUN \
   cd .. && \
   rm -rf patchelf-0.10
 
+# Build Python 3.7
+# —-enable-optimizations for a stable/release build
+RUN \
+  cd ${build_dir} && \
+  wget --quiet -O - https://www.python.org/ftp/python/3.7.9/Python-3.7.9.tgz | tar xzf - && \
+  cd Python-3.7.9/ && \  
+  ./configure --enable-optimizations && \ 
+  make altinstall && \
+  cd .. && rm -rf Python-3.7.9
+
+# Install Python dependencies 
+RUN \
+  /usr/local/bin/pip3.7 install numpy ninja pyyaml mkl mkl-include setuptools cmake cffi typing_extensions future six requests dataclasses  
+
+# Install CMake
+# v3.5 minimum is required
+ARG cmake_install_dir=/usr/local/cmake
+RUN \
+  cd ${build_dir} && \
+  wget --quiet https://github.com/Kitware/CMake/releases/download/v3.19.3/cmake-3.19.3-Linux-x86_64.sh && \
+  chmod +x cmake-3.19.3-Linux-x86_64.sh && \
+  mkdir ${cmake_install_dir} && \
+  ./cmake-3.19.3-Linux-x86_64.sh --skip-license --prefix=${cmake_install_dir} && \
+  rm -f cmake-3.19.3-Linux-x86_64.sh
+
+# Add cmake to PATH
+ENV PATH=${PATH}:${cmake_install_dir}/bin
+
+# Clone PyTorch
+RUN \
+  cd ${build_dir} && \
+  git -c advice.detachedHead=false clone --depth=1 --branch=v1.7.1 https://github.com/pytorch/pytorch.git && \
+  cd pytorch && \
+  git submodule sync && \
+  git submodule update --init --recursive 
+
+# Build LibTorch
+RUN \
+  cd ${build_dir}/pytorch && \
+  /usr/local/bin/python3.7 setup.py install && \
```

```suggestion
  export USE_DISTRIBUTED=OFF
  /usr/local/bin/python3.7 setup.py install && \
```

This needs to be explicitly specified on Linux, whereas it's the default on macOS.